res/skins/QMLDemo/WaveformShader.qml

```diff
@@ -0,0 +1,222 @@
+import Mixxx 0.1 as Mixxx
+import QtQuick 2.12
+
+ShaderEffect {
+    id: root
+
+    property string group // required
+    property var deckPlayer: Mixxx.PlayerManager.getPlayer(group)
+    property variant framebufferSize: Qt.size(width, height)
+    property int waveformLength: root.deckPlayer.waveformLength
+    property int textureSize: root.deckPlayer.waveformTextureSize
+    property int textureStride: root.deckPlayer.waveformTextureStride
+    property real firstVisualIndex: 1
+    property real lastVisualIndex: root.deckPlayer.waveformLength / 2
+    property color axesColor: "#FFFFFF"
+    property color highColor: "#FF0000"
+    property color midColor: "#00FF00"
+    property color lowColor: "#0000FF"
+    property real highGain: filterWaveformEnableControl.value ? (filterHighKillControl.value ? 0 : filterHighControl.value) : 1
+    property real midGain: filterWaveformEnableControl.value ? (filterMidKillControl.value ? 0 : filterMidControl.value) : 1
+    property real lowGain: filterWaveformEnableControl.value ? (filterLowKillControl.value ? 0 : filterLowControl.value) : 1
+    property real allGain: pregainControl.value
+    property variant waveformTexture
+
```

Comment:
are you sure we don't need to supply a vertex shader too? Even though it doesn't do much in reality.